pkg/collector/corechecks/net/network.go

```diff
@@ -0,0 +1,312 @@
+	tags := []string{fmt.Sprintf("device:%s", interfaceIO.Name)}
+	sender.Rate("system.net.bytes_rcvd", float64(interfaceIO.BytesRecv), "", tags)
+	sender.Rate("system.net.bytes_sent", float64(interfaceIO.BytesSent), "", tags)
+	sender.Rate("system.net.packets_in.count", float64(interfaceIO.PacketsRecv), "", tags)
+	sender.Rate("system.net.packets_in.error", float64(interfaceIO.Errin), "", tags)
+	sender.Rate("system.net.packets_out.count", float64(interfaceIO.PacketsSent), "", tags)
+	sender.Rate("system.net.packets_out.error", float64(interfaceIO.Errout), "", tags)
+}
+
+func submitProtocolMetrics(sender aggregator.Sender, protocolStats net.ProtoCountersStat) {
+	if protocolMapping, ok := protocolsMetricsMapping[protocolStats.Protocol]; ok {
+		for rawMetricName, metricName := range protocolMapping {
+			if metricValue, ok := protocolStats.Stats[rawMetricName]; ok {
+				sender.Rate(metricName, float64(metricValue), "", []string{})
+				sender.MonotonicCount(fmt.Sprintf("%s.count", metricName), float64(metricValue), "",
+					[]string{})
+			}
+		}
+	}
+}
+
+func submitConnectionsMetrics(sender aggregator.Sender, protocolName string, stateMetricSuffixMapping map[string]string, connectionsStats []net.ConnectionStat) {
+	metricCount := map[string]float64{}
+	for _, suffix := range stateMetricSuffixMapping {
+		metricCount[suffix] = 0
+	}
+
+	for _, connectionStats := range connectionsStats {
+		metricCount[stateMetricSuffixMapping[connectionStats.Status]]++
+	}
+
+	for suffix, count := range metricCount {
+		sender.Gauge(fmt.Sprintf("system.net.%s.%s", protocolName, suffix),
+			count, "", []string{})
+	}
+}
+
+func netstatTCPExtCounters() (map[string]int64, error) {
+
+	f, err := os.Open("/proc/net/netstat")
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	counters := map[string]int64{}
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		i := strings.IndexRune(line, ':')
+		if i == -1 {
+			return nil, errors.New("/proc/net/netstat is not fomatted correctly, expected ':'")
+		}
+		proto := strings.ToLower(line[:i])
+		if proto != "tcpext" {
+			continue
+		}
+
+		counterNames := strings.Split(line[i+2:], " ")
+
+		if !scanner.Scan() {
+			return nil, errors.New("/proc/net/netstat is not fomatted correctly, not data line")
+		}
+		line = scanner.Text()
+
+		counterValues := strings.Split(line[i+2:], " ")
+		if len(counterNames) != len(counterValues) {
+			return nil, errors.New("/proc/net/netstat is not fomatted correctly, expected same number of columns")
+		}
+
+		for j := range counterNames {
+			value, err := strconv.ParseInt(counterValues[j], 10, 64)
+			if err != nil {
+				return nil, err
+			}
+			counters[counterNames[j]] = value
+		}
+	}
+
+	return counters, nil
+}
+
+// Configure configures the network checks
+func (c *NetworkCheck) Configure(rawInstance integration.Data, rawInitConfig integration.Data) error {
+	err := yaml.Unmarshal(rawInitConfig, &c.config.initConf)
```

Comment:
Can you please call `CheckBase.CommonConfigure` to handle the common options, see https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/corechecks/checkbase.go#L65